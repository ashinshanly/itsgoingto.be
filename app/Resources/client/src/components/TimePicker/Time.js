import React from 'react'
import PropTypes from 'prop-types'

class Time extends React.Component {
  constructor (props) {
    super(props)
  }

  handleClickOutside = (event) => {
    this.props.onClickOutside(event)
  }

  render () {
    return (
      <div>
        Time picker popup content
      </div>
    )
  }
}

Time.propTypes = {
}

Time.defaultProps = {
}

export default Time
