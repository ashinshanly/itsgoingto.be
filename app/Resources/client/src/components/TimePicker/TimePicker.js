import React from 'react'
import PopperComponent, { popperPlacementPositions } from './PopperComponent'
import classnames from 'classnames'

const outsideClickIgnoreClass = 'react-timepicker-ignore-onclickoutside'

class TimePicker extends React.Component {
  constructor (props) {
    super(props)
    this.state = { open: false }
  }

  setOpen = (open) => {
    this.setState({
      open: open
    })
  }

  handleChange = (event) => {
  }

  handleFocus = (event) => {
    if (!this.state.preventFocus) {
      this.setOpen(true)
    }
  }

  onInputClick = () => {
    if (!this.props.disabled) {
      this.setOpen(true)
    }
  }

  renderInput = () => {
    var className = classnames(this.props.className, {
      [outsideClickIgnoreClass]: this.state.open
    })

    const customInput = this.props.customInput || <input type="text" />
    const inputValue = this.props.value

    return React.cloneElement(customInput, {
      ref: 'input',
      value: inputValue,
      onChange: this.handleChange,
      onClick: this.onInputClick,
      onFocus: this.handleFocus,
      id: this.props.id,
      name: this.props.name,
      placeholder: this.props.placeholderText,
      disabled: this.props.disabled,
      className: className,
      readOnly: this.props.readOnly
    })
  }

  render () {
    return (
      <PopperComponent
        hidePopper={(!this.state.open || this.props.disabled)}
        targetComponent={
          <div className="react-timepicker__input-container">
            {this.renderInput()}
          </div>
        }
        popperComponent={<div><h1>Hii</h1></div>}/>
    )
  }
}

TimePicker.propTypes = {

}

export default TimePicker
