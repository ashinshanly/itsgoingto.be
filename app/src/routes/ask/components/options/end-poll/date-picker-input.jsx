import React from 'react'
import PropTypes from 'prop-types'

export class DatePickerInput extends React.Component {
  render = () => (
    <span
      className='input-option-datepicker-label'
      onClick={this.props.onClick}>
      {this.props.value}
    </span>
  )
}

DatePickerInput.propTypes = {
  onClick : PropTypes.func,
  value   : PropTypes.string
}

export default DatePickerInput
