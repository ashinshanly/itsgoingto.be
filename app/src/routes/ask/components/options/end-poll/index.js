import EndPoll from './end-poll'

export default EndPoll
