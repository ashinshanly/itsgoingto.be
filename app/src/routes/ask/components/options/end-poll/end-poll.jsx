import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import moment from 'moment'
import Slider from 'react-rangeslider'
import DatePicker from 'react-datepicker'
import TimePicker from 'antd/lib/time-picker'

import { updatePoll } from 'store/poll'
import DatePickerInput from './date-picker-input'

import 'react-rangeslider/lib/index.css'
import 'react-datepicker/dist/react-datepicker-cssmodules.css'
import 'antd/lib/time-picker/style/index.css'
import './end-poll.scss'

const formatEndin = (value) => value + ' hour' + (value > 1 ? 's' : '')

class EndPoll extends React.Component {
  componentDidUpdate = () => {
    const { poll } = this.props

    if (typeof poll.endIn === 'undefined') {
      this.props.updateOptions({
        identifier : '',
        endIn      : 1
      })
    }
    if (typeof poll.endAt === 'undefined') {
      this.props.updateOptions({
        identifier : '',
        endAt      : moment().add(1, 'days').seconds(0).milliseconds(0)
      })
    }
  }

  handleEndTypeChange = (event) =>
    this.props.updateOptions({
      identifier : '',
      endType    : event.target.value
    })

  handleEndInChange = (value) =>
    this.props.updateOptions({
      identifier : '',
      endIn      : value
    })

  handleEndAtChange = (value) => {
    const minDate = moment().add(1, 'hour').seconds(0).milliseconds(0)

    if (value.isBefore(minDate)) {
      this.props.updateOptions({
        identifier : '',
        endAt      : minDate
      })
    } else {
      this.props.updateOptions({
        identifier : '',
        endAt      : value
      })
    }
  }

  timePickerDisabledHours = () => {
    const { poll } = this.props

    if (poll.endAt.isSame(moment(), 'day')) {
      return [...Array(moment().hour() + 1).keys()]
    }

    return []
  }

  timePickerDisabledMinutes = () => {
    const { poll } = this.props

    if (poll.endAt.isSame(moment().add(1, 'hour'), 'hour')) {
      return [...Array(moment().minute()).keys()]
    }

    return []
  }

  render () {
    const { poll } = this.props

    return (
      <div className='input-option input-option-endPoll'>
        <div className='input-option-radios'>
          <input
            id='end-never'
            className='input-radio input-checkbox-endType'
            name='endType'
            type='checkbox'
            checked={poll.endType !== 'endAt' && poll.endType !== 'endIn'}
            value='endNever'
            onChange={this.handleEndTypeChange} />
          <label
            htmlFor='end-never'
            className='input-label input-label-options input-label-endType'>
            {'Don\'t End Poll'}
          </label>
          <input
            id='end-at'
            className='input-radio input-checkbox-endType'
            name='endType'
            type='checkbox'
            checked={poll.endType === 'endAt'}
            value='endAt'
            onChange={this.handleEndTypeChange} />
          <label
            htmlFor='end-at'
            className='input-label input-label-options input-label-endType'>
            End Poll At
          </label>
          <input
            id='end-in'
            className='input-radio input-checkbox-endType'
            name='endType'
            type='radio'
            value='endIn'
            checked={poll.endType === 'endIn'}
            onChange={this.handleEndTypeChange} />
          <label
            htmlFor='end-in'
            className='input-label input-label-options input-label-endType'>
            End Poll In
          </label>
        </div>
        {poll.endType === 'endAt' &&
          <div className='input-option-datepicker'>
            <DatePicker
              customInput={<DatePickerInput />}
              dateFormat='DD/MM/YYYY'
              minDate={moment()}
              selected={poll.endAt}
              onChange={this.handleEndAtChange} />
            <TimePicker
              format='HH:mm'
              disabledHours={this.timePickerDisabledHours}
              disabledMinutes={this.timePickerDisabledMinutes}
              hideDisabledOptions
              value={poll.endAt}
              onChange={this.handleEndAtChange} />
          </div>
        }
        {poll.endType === 'endIn' &&
          <div className='input-option-slider'>
            <div className='input-option-slider-label'>{formatEndin(poll.endIn)}</div>
            <Slider
              min={1}
              max={24}
              value={poll.endIn}
              tooltip={false}
              onChange={this.handleEndInChange}
            />
          </div>
        }
      </div>
    )
  }
}

EndPoll.propTypes = {
  poll          : PropTypes.object.isRequired,
  updateOptions : PropTypes.func.isRequired
}

const mapDispatchToProps = (dispatch) => ({
  updateOptions: (value) => dispatch(updatePoll(value))
})

export default connect(null, mapDispatchToProps)(EndPoll)
