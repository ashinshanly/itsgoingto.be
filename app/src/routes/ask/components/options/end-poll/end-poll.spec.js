/* global expect, jest */
import React from 'react'
import { shallow, mount } from 'enzyme'
import moment from 'moment'
import Slider from 'react-rangeslider'
import DatePicker from 'react-datepicker'
import TimePicker from 'antd/lib/time-picker'
import { EndPoll } from './end-poll'

Date.now = jest.fn(() => new Date(Date.UTC(2018, 2, 10, 10, 30, 0)).valueOf())

const props = {
  poll          : { endType: 'endNever' },
  updateOptions : jest.fn(),
}

describe('(Route) Ask', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  describe('(Component) EndPoll', () => {
    describe('(Action) onChange', () => {
      describe('input endNever', () => {
        it('should call prop updateOptions', () => {
          const wrapper = mount(<EndPoll {...props} />)

          wrapper.find('#end-never').simulate('change')
          expect(props.updateOptions).toHaveBeenCalledWith({
            identifier : '',
            endType    : 'endNever',
          })
        })
      })

      describe('input endAt', () => {
        it('should call prop updateOptions', () => {
          const wrapper = mount(<EndPoll {...props} />)

          wrapper.find('#end-at').simulate('change')
          expect(props.updateOptions).toHaveBeenCalledWith({
            identifier : '',
            endType    : 'endAt',
          })
        })
      })

      describe('input endIn', () => {
        it('should call prop updateOptions', () => {
          const wrapper = mount(<EndPoll {...props} />)

          wrapper.find('#end-in').simulate('change')
          expect(props.updateOptions).toHaveBeenCalledWith({
            identifier : '',
            endType    : 'endIn',
          })
        })
      })

      describe('endAt DatePicker', () => {
        it('should call prop updateOptions', () => {
          const date = moment('2018-03-12')
          const wrapper = shallow(<EndPoll {...props} poll={{ endType: 'endAt', endAt: date }} />)

          wrapper.find(DatePicker).simulate('change', date)
          expect(props.updateOptions).toHaveBeenCalledWith({
            identifier : '',
            endAt      : date,
          })
        })
      })

      describe('endAt TimePicker', () => {
        it('should call prop updateOptions with date', () => {
          const date = moment('2018-03-12 10:30:00')
          const wrapper = shallow(<EndPoll {...props} poll={{ endType: 'endAt', endAt: date }} />)

          wrapper.find(TimePicker).simulate('change', date)
          expect(props.updateOptions).toHaveBeenCalledWith({
            identifier : '',
            endAt      : date,
          })
        })

        it('should call prop updateOptions with min date', () => {
          const date = moment('2018-03-10 10:45:00')
          const wrapper = shallow(<EndPoll {...props} poll={{ endType: 'endAt', endAt: date }} />)

          wrapper.find(TimePicker).simulate('change', date)
          expect(props.updateOptions).toHaveBeenCalledWith({
            identifier : '',
            endAt      : expect.any(moment), // One day figure out how to properly test this value
          })
        })
      })

      describe('endIn Slider', () => {
        it('should call prop updateOptions', () => {
          const wrapper = shallow(<EndPoll {...props} poll={{ endType: 'endIn', endIn: 5 }} />)

          wrapper.find(Slider).simulate('change', 6)
          expect(props.updateOptions).toHaveBeenCalledWith({
            identifier : '',
            endIn      : 6,
          })
        })
      })
    })

    describe('(Render)', () => {
      it('matches snapshot', () => {
        const wrapper = shallow(<EndPoll {...props} />)

        expect(wrapper).toMatchSnapshot()
      })

      describe('when endType is endAt', () => {
        it('matches snapshot', () => {
          const wrapper = shallow(<EndPoll {...props} poll={{ endType: 'endAt' }} />)

          expect(wrapper).toMatchSnapshot()
        })
      })

      describe('when endType is endIn', () => {
        it('matches snapshot', () => {
          const wrapper = shallow(<EndPoll {...props} poll={{ endType: 'endIn' }} />)

          expect(wrapper).toMatchSnapshot()
        })
      })
    })
  })
})