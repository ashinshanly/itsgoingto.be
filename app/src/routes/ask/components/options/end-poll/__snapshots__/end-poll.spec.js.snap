// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Route) Ask (Component) EndPoll (Render) matches snapshot 1`] = `
<div
  className="input-option input-option-endPoll"
>
  <div
    className="input-option-radios"
  >
    <input
      checked={true}
      className="input-radio input-checkbox-endType"
      id="end-never"
      name="endType"
      onChange={[Function]}
      type="checkbox"
      value="endNever"
    />
    <label
      className="input-label input-label-options input-label-endType"
      htmlFor="end-never"
    >
      Don't End Poll
    </label>
    <input
      checked={false}
      className="input-radio input-checkbox-endType"
      id="end-at"
      name="endType"
      onChange={[Function]}
      type="checkbox"
      value="endAt"
    />
    <label
      className="input-label input-label-options input-label-endType"
      htmlFor="end-at"
    >
      End Poll At
    </label>
    <input
      checked={false}
      className="input-radio input-checkbox-endType"
      id="end-in"
      name="endType"
      onChange={[Function]}
      type="radio"
      value="endIn"
    />
    <label
      className="input-label input-label-options input-label-endType"
      htmlFor="end-in"
    >
      End Poll In
    </label>
  </div>
</div>
`;

exports[`(Route) Ask (Component) EndPoll (Render) when endType is endAt matches snapshot 1`] = `
<div
  className="input-option input-option-endPoll"
>
  <div
    className="input-option-radios"
  >
    <input
      checked={false}
      className="input-radio input-checkbox-endType"
      id="end-never"
      name="endType"
      onChange={[Function]}
      type="checkbox"
      value="endNever"
    />
    <label
      className="input-label input-label-options input-label-endType"
      htmlFor="end-never"
    >
      Don't End Poll
    </label>
    <input
      checked={true}
      className="input-radio input-checkbox-endType"
      id="end-at"
      name="endType"
      onChange={[Function]}
      type="checkbox"
      value="endAt"
    />
    <label
      className="input-label input-label-options input-label-endType"
      htmlFor="end-at"
    >
      End Poll At
    </label>
    <input
      checked={false}
      className="input-radio input-checkbox-endType"
      id="end-in"
      name="endType"
      onChange={[Function]}
      type="radio"
      value="endIn"
    />
    <label
      className="input-label input-label-options input-label-endType"
      htmlFor="end-in"
    >
      End Poll In
    </label>
  </div>
  <div
    className="input-option-datepicker"
  >
    <t
      allowSameDay={false}
      customInput={<DatePickerInput />}
      dateFormat="DD/MM/YYYY"
      dateFormatCalendar="MMMM YYYY"
      disabled={false}
      disabledKeyboardNavigation={false}
      dropdownMode="scroll"
      minDate={"2018-03-10T10:30:00.000Z"}
      monthsShown={1}
      onBlur={[Function]}
      onChange={[Function]}
      onClickOutside={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMonthChange={[Function]}
      onSelect={[Function]}
      shouldCloseOnSelect={true}
      utcOffset={-0}
      withPortal={false}
    />
    <TimePicker
      align={
        Object {
          "offset": Array [
            0,
            -2,
          ],
        }
      }
      disabled={false}
      disabledHours={[Function]}
      disabledMinutes={[Function]}
      focusOnOpen={true}
      format="HH:mm"
      hideDisabledOptions={true}
      onChange={[Function]}
      placement="bottomLeft"
      prefixCls="ant-time-picker"
      transitionName="slide-up"
    />
  </div>
</div>
`;

exports[`(Route) Ask (Component) EndPoll (Render) when endType is endIn matches snapshot 1`] = `
<div
  className="input-option input-option-endPoll"
>
  <div
    className="input-option-radios"
  >
    <input
      checked={false}
      className="input-radio input-checkbox-endType"
      id="end-never"
      name="endType"
      onChange={[Function]}
      type="checkbox"
      value="endNever"
    />
    <label
      className="input-label input-label-options input-label-endType"
      htmlFor="end-never"
    >
      Don't End Poll
    </label>
    <input
      checked={false}
      className="input-radio input-checkbox-endType"
      id="end-at"
      name="endType"
      onChange={[Function]}
      type="checkbox"
      value="endAt"
    />
    <label
      className="input-label input-label-options input-label-endType"
      htmlFor="end-at"
    >
      End Poll At
    </label>
    <input
      checked={true}
      className="input-radio input-checkbox-endType"
      id="end-in"
      name="endType"
      onChange={[Function]}
      type="radio"
      value="endIn"
    />
    <label
      className="input-label input-label-options input-label-endType"
      htmlFor="end-in"
    >
      End Poll In
    </label>
  </div>
  <div
    className="input-option-slider"
  >
    <div
      className="input-option-slider-label"
    >
      undefined hour
    </div>
    <Slider
      handleLabel=""
      labels={Object {}}
      max={24}
      min={1}
      onChange={[Function]}
      orientation="horizontal"
      reverse={false}
      step={1}
      tooltip={false}
      value={0}
    />
  </div>
</div>
`;
